File: ios/ArtsyWidget/Models/Artwork.swift
import Foundation
import UIKit

struct Artwork: Codable {
    static func fallback() -> Artwork {
        return Fixtures.primaryArtwork
    }
    
    let artist: Artist
    let artworkImages: [ArtworkImage]
    let id: String
    let title: String
    
    var image: UIImage?
    
    var firstImageToken: String {
        let defaultImage = artworkImages.first(where: { $0.isDefault })
        let sortedImages = artworkImages.sorted { $0.position < $1.position }
        let firstImage = defaultImage ?? sortedImages.first ?? ArtworkImage.fallback()
        
        return firstImage.geminiToken
    }
    
    var url: URL {
        let link = ["https://www.artsy.net/artwork/", id].joined()
        return WidgetUrl.from(link: link)!
    }
    
    enum CodingKeys: String, CodingKey {
        case artworkImages = "images"
        
        case artist
        case id
        case title
    }
}

File: android/app/src/main/java/net/artsy/app/widgets/models/Artwork.kt
package net.artsy.app.widgets.models

import android.graphics.Bitmap

data class Artwork(
    val id: String,
    val title: String,
    val artist: Artist,
    val artworkImages: List<ArtworkImage>,
    var image: Bitmap? = null
) {
    val firstImageToken: String
        get() {
            val defaultImage = artworkImages.find { it.isDefault }
            val sortedImages = artworkImages.sortedBy { it.position }
            val firstImage = defaultImage ?: sortedImages.firstOrNull() ?: ArtworkImage.fallback()
            return firstImage.geminiToken
        }

    val url: String
        get() = "https://www.artsy.net/artwork/$id?utm_medium=android_widget"

    companion object {
        fun fallback(): Artwork {
            return Artwork(
                id ="alex-katz-brisk-day-34",
                title = "Brisk Day",
                artist = Artist.fallback(),
                artworkImages = listOf(ArtworkImage.fallback())
            )
        }
    }
}